vertx-sql-client/src/main/java/io/vertx/sqlclient/Tuple.java

```diff
@@ -645,20 +654,12 @@ default BigDecimal getBigDecimal(int pos) {
   @GenIgnore(GenIgnore.PERMITTED_TYPE)
   default Boolean[] getArrayOfBooleans(int pos) {
     Object val = getValue(pos);
-    if (val == null) {
-      return null;
-    } else if (val instanceof Boolean[]) {
-      return (Boolean[]) val;
-    } else if (val.getClass() == Object[].class) {
-      Object[] array = (Object[]) val;
-      Boolean[] booleanArray = new Boolean[array.length];
-      for (int i = 0;i < array.length;i++) {
-        booleanArray[i] = (Boolean) array[i];
-      }
-      return booleanArray;
-    } else {
-      return (Boolean[]) val; // Throw CCE
+    if (val == null) return null;
+    if (val instanceof Boolean[]) return (Boolean[]) val;
+    if (val.getClass().isArray()) {
```

Comment:
we could have a top level `if` that checks if `val` is an array, and then other tests inside the block which inspect the `componentType`

Reply:
This applies to the other, similar changes